### PR TITLE
Adding documentation for Advanced Provenance Queries

### PR DESCRIPTION
**Pull request by Rosspet**
# Provena PR (Minor): Adding documentation for Advanced Provenance Queries

## Description

Adding documentation for advanced provenance queries. Before, merging, please serve the documentation and ensure the images are displaying appropriately.

### CODE REVIEW

File: docs/provenance/exploring-provenance/exploring-record-lineage.md [hunk 1]
```diff
@@ -245,6 +245,28 @@ To use the "View Selected Entity" button, first select the node to show it's [de
 | :------------------------------------------------------------------------------------------------------: |
 | <img src="../../assets/images/provenance/exploring/graph/view_entity.png" alt="drawing" width="800"/> |
 
+
+### Advanced Provenance Queries
+
+In addition to interrogation of provenance via normal queries facilitated by the manual expansion of entities, the Provenance System supplies special querying capabilities for automatically discerning affected downstream or upstream Datasets and Agents. These special queries can explore at a greater depth, and filter by relationships and types. There are four types of special queries:
```
Comment (PeterBaker0) [inline docs/provenance/exploring-provenance/exploring-record-lineage.md:251]: I think this wording could be simplified. Particularly this bit "interrogation of provenance via normal queries facilitated by the manual expansion of entities, the Provenance System supplies special querying capabilities for automatically discerning affected downstream or upstream Datasets and Agents"

Also maybe we should be consistent re: "advanced" vs "special" queries - I don't mind which.

File: docs/provenance/exploring-provenance/exploring-record-lineage.md [hunk 2]
```diff
@@ -245,6 +245,28 @@ To use the "View Selected Entity" button, first select the node to show it's [de
...
+
+**Affected Downstream Datasets**: Expands all nodes in the downstream direction to find any affected Datasets.  
+**Contributing Upstream Datasets**: Expands all nodes in the upstream direction to find any contributing Datasets.  
+**Affected Downstream Agents**: Expands all nodes relationships in the downstream direction to find any affected Agents.  
+**Contributing Upstream Agents**: Expands all nodes relationships in the upstream direction to find any contributing Agents.  
+
+#### How to Perform a Special Query
+
+These queries can be made from a starting node by single clicking and using the "Provenance Queries" drop down (which appears on the right panel) to select the desired query. For example, selecting a Dataset used as Input to a Model Run, and querying for "Contributing Upstream Datasets" will return all the upstream lineage of connected datasets that were (by definition of upstream) occuring before, or used to produce or contribute towards this input dataset. The below images illustrate how to perform this activity in the Provenance System.
```
Comment (PeterBaker0) [inline docs/provenance/exploring-provenance/exploring-record-lineage.md:260]: I think this sentence could be simplified

> will return all the upstream lineage of connected datasets that were (by definition of upstream) occuring before, or used to produce or contribute towards this input dataset.

Maybe just "will return all upstream datasets which were used to produce, or contributed to the production of the selected dataset."

**Review by PeterBaker0 — CHANGES_REQUESTED** [review]
Couple of minor wording changes but looks good 👍

**Comment by PeterBaker0** [comment]
I'll pick up on this and close this out as Ross is unavailable until later next week

**Review by PeterBaker0 — APPROVED** [review]
Simplified wording and cleaned up old artifacts from previous docs.

**Comment by jyucsiro** [comment]
nice work!

**Pull request merged**